backend/index.js

```diff
@@ -18,7 +18,23 @@ try {
 const app = express();
 app.disable('x-powered-by');
 app.use(express.raw({ type: '*/*', limit: '100mb' }));
-if (gzipEnabled) app.use(compression());
+if (gzipEnabled)
+  app.use(
+    compression({
+      filter: (req, res) => {
+        //TODO change it to a custom header, pass the header from the front
+        if (
+          req.originalUrl.includes('watch=') ||
+          req.originalUrl.includes('follow=')
+        ) {
```

Comment:
Since we're only proxing k8s requests, I'd say we don't even need a custom header - we can take it from query params instead of an entire URL.